src/samplers/multinomial.jl

```diff
@@ -1,33 +1,33 @@
-function multinom_rand!(rng::AbstractRNG, n::Int, p::AbstractVector{Float64},
-                         x::AbstractVector{<:Real})
+function multinom_rand!(rng::AbstractRNG, n::Int, p::AbstractVector{Tp},
+                         x::AbstractVector{Tx}) where {Tx <: Real, Tp <: Real}
     k = length(p)
     length(x) == k || throw(DimensionMismatch("Invalid argument dimension."))
 
-    rp = 1.0  # remaining total probability
+    rp = Tp(1.0)  # remaining total probability
     i = 0
     km1 = k - 1
 
     while i < km1 && n > 0
         i += 1
         @inbounds pi = p[i]
         if pi < rp
-            xi = rand(rng, Binomial(n, pi / rp))
+            xi = Tx(rand(rng, Binomial(n, Float64(pi / rp))))
```

Comment:
This is not needed, updating `x[i]` in the line below will convert to the correct type anyway; additionally, the constructor of `Binomial` (should) take care of promotions/conversions (and we just continue with using the type of `p`/`rp`):
```suggestion
            xi = rand(rng, Binomial(n, pi / rp))
```